### PR TITLE
`azurerm_windows_web_app`, `azurerm_windows_web_app_slot`, `azurerm_windows_function_app` - fix dotnet runtime setting

### PR DESCRIPTION
**Pull request by xiaxyi**
Correct windows app service dotnet app stack settings:
1. v3.0 is not a supported value for dotnet stack, thus, we need to remove it in 4.0 provider.
2. Remove dotnet core support in terraform as it was retried by api :https://github.com/azure-deprecation/dashboard/issues/216
4. correct existed dotnet core app stack settings.

acc tests:
```
--- PASS: TestAccWindowsWebApp_withDotNetCore (365.18s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    365.886s

--- PASS: TestAccWindowsWebAppSlot_withDotNetCore (590.03s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice    590.789s


```

user reported issues:
fix https://github.com/hashicorp/terraform-provider-azurerm/issues/16867

### CODE REVIEW

File: internal/services/appservice/helpers/web_app_slot_schema.go [hunk 1]
```diff
@@ -1075,10 +1081,16 @@ func FlattenSiteConfigWindowsAppSlot(appSiteSlotConfig *web.SiteConfig, currentS
 
 	winAppStack := ApplicationStackWindows{}
 
-	winAppStack.NetFrameworkVersion = pointer.From(appSiteSlotConfig.NetFrameworkVersion)
-	if currentStack == CurrentStackDotNetCore {
-		winAppStack.NetCoreVersion = pointer.From(appSiteSlotConfig.NetFrameworkVersion)
+	if appSiteSlotConfig.NetFrameworkVersion != nil {
+		winAppStack.NetFrameworkVersion = pointer.From(appSiteSlotConfig.NetFrameworkVersion)
+		if *appSiteSlotConfig.NetFrameworkVersion == DotNetCLRVersionFourPointOh {
+			if !features.FourPointOhBeta() && metadata.ResourceData.Get("site_config.0.application_stack.0.dotnet_core_version").(string) != "" {
```
Comment (jackofallops) [inline internal/services/appservice/helpers/web_app_slot_schema.go:1087]: I think we can do this without the metadata?In the pre-4.0 path we'll still have `currentStack == CurrentStackDotNetCore` to check this? 

e.g. 
```suggestion
			if !features.FourPointOhBeta() && currentStack == CurrentStackDotNetCore {
```

File: internal/services/appservice/helpers/web_app_slot_schema.go [hunk 2]
```diff
@@ -1021,7 +1027,7 @@ func ExpandSiteConfigWindowsWebAppSlot(siteConfig []SiteConfigWindowsWebAppSlot,
 	return expanded, &currentStack, nil
 }
 
-func FlattenSiteConfigWindowsAppSlot(appSiteSlotConfig *web.SiteConfig, currentStack string, healthCheckCount *int) []SiteConfigWindowsWebAppSlot {
+func FlattenSiteConfigWindowsAppSlot(appSiteSlotConfig *web.SiteConfig, currentStack string, healthCheckCount *int, metadata sdk.ResourceMetaData) []SiteConfigWindowsWebAppSlot {
```
Comment (jackofallops) [inline internal/services/appservice/helpers/web_app_slot_schema.go:1030]: I don't believe we need to pass metadata around here, and we shouldn't use d.Get in most cases in a `Read`, see below
```suggestion
func FlattenSiteConfigWindowsAppSlot(appSiteSlotConfig *web.SiteConfig, currentStack string, healthCheckCount *int) []SiteConfigWindowsWebAppSlot {
```

File: internal/services/appservice/helpers/windows_web_app_schema.go [hunk 3]
```diff
@@ -654,7 +660,7 @@ func ExpandSiteConfigWindows(siteConfig []SiteConfigWindows, existing *web.SiteC
 	return expanded, &currentStack, nil
 }
 
-func FlattenSiteConfigWindows(appSiteConfig *web.SiteConfig, currentStack string, healthCheckCount *int) ([]SiteConfigWindows, error) {
+func FlattenSiteConfigWindows(appSiteConfig *web.SiteConfig, currentStack string, healthCheckCount *int, metadata sdk.ResourceMetaData) ([]SiteConfigWindows, error) {
```
Comment (jackofallops) [inline internal/services/appservice/helpers/windows_web_app_schema.go:663]: As above on not passing metadata.

File: internal/services/appservice/helpers/windows_web_app_schema.go [hunk 4]
```diff
@@ -710,12 +716,16 @@ func FlattenSiteConfigWindows(appSiteConfig *web.SiteConfig, currentStack string
...
 
 	var winAppStack ApplicationStackWindows
-	if currentStack == CurrentStackDotNetCore {
-		winAppStack.NetCoreVersion = pointer.From(appSiteConfig.NetFrameworkVersion)
-	}
-	if currentStack == CurrentStackDotNet {
+	if appSiteConfig.NetFrameworkVersion != nil {
 		winAppStack.NetFrameworkVersion = pointer.From(appSiteConfig.NetFrameworkVersion)
+		if *appSiteConfig.NetFrameworkVersion == DotNetCLRVersionFourPointOh {
+			if !features.FourPointOhBeta() && metadata.ResourceData.Get("site_config.0.application_stack.0.dotnet_core_version").(string) != "" {
```
Comment (jackofallops) [inline internal/services/appservice/helpers/windows_web_app_schema.go:722]: as above wrt checking `currentStack` here also.

**Review by jackofallops — CHANGES_REQUESTED** [review]
Thanks @xiaxyi - apologies for the delayed review. I've left a couple points below to check / adjust if you can take a look, then I'll re-review.

Thanks!

File: internal/services/appservice/helpers/web_app_slot_schema.go (hunk 1 above)
Comment (xiaxyi) [reply internal/services/appservice/helpers/web_app_slot_schema.go:1087]: updated

File: internal/services/appservice/helpers/windows_web_app_schema.go (hunk 3 above)
Comment (xiaxyi) [reply internal/services/appservice/helpers/windows_web_app_schema.go:663]: updated

File: internal/services/appservice/helpers/windows_web_app_schema.go (hunk 4 above)
Comment (xiaxyi) [reply internal/services/appservice/helpers/windows_web_app_schema.go:722]: updated

**Comment by xiaxyi** [comment]
Thanks @jackofallops , updated the code.

**Pull request closed without merging**